dbusmock/__main__.py

```diff
@@ -117,4 +117,7 @@ def parse_args():
         main_object.AddTemplate(args.template, parameters)
 
     dbusmock.mockobject.objects[args.path] = main_object
-    main_loop.run()
+
+    context = GLib.MainContext.default()
+    while should_run:
```

Comment:
`should_run` is never changed; that's normally done through g_main_loop_quit(), except that nothing in our mocks calls that (a mock is simply terminated). So why not a simple `while True:` here?

Reply:
`should_run` is absolutely changed via `should_run.pop` as the signal receiver for `Disconnected`.

Reply:
Argh, of course, sorry.